### PR TITLE
Added per item highlighting/hiding to GroundItems

### PR DESCRIPTION
**Pull request by abelbriggs1**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsConfig.java [hunk 1]
```diff
@@ -58,10 +58,30 @@ default boolean showGEPrice()
...
 	{
 		return false;
 	}
+
+	@ConfigItem(
+		keyName = "hiddenItems",
+		name = "Hidden Items",
+		description = "Configures hidden ground items. Format: (item), (item)"
+	)
+	default String getHiddenItems()
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsConfig.java]: Does this work okay if it is just ```String getHiddenItems()```?

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsOverlay.java [hunk 2]
```diff
@@ -149,15 +161,20 @@ public Dimension render(Graphics2D graphics)
 					Item item = (Item) current;
 					int itemId = item.getId();
 					int itemQuantity = item.getQuantity();
+					// Item does not support getting the name, so we need to get the ItemComposition instead
+					ItemComposition itemDefinition = client.getItemDefinition(itemId);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsOverlay.java]: The client has a cache for these but only for 64 items. Maybe we should cache this ourselves.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsOverlay.java [hunk 3]
```diff
@@ -149,15 +161,20 @@ public Dimension render(Graphics2D graphics)
...
 
 					Integer currentQuantity = items.get(itemId);
-					if (currentQuantity == null)
-					{
-						items.put(itemId, itemQuantity);
-					}
-					else
+					if (!hiddenItems.contains(itemDefinition.getName()))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsOverlay.java]: Why not just:

```java
if (hiddenItems.contains(itemDefinition.getName())) continue;
```

Comment (abelbriggs1) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsOverlay.java]: Using continue there breaks the client, it never advances to the next item to put into the arraylist.

artificially throwing a current = current.getNext() and then continuing would probably work though.

**Pull request merged**